03-streams/02-line-split-stream/LineSplitStream.js

```diff
@@ -4,12 +4,35 @@ const os = require('os');
 class LineSplitStream extends stream.Transform {
   constructor(options) {
     super(options);
+    this.chunkToSend = '';
   }
 
-  _transform(chunk, encoding, callback) {
+  _transform(chunk, encoding, done) {
+    if (!chunk.toString().includes(os.EOL)) {
+      this.chunkToSend += chunk;
+      done();
+    } else {
+      const splittedChunk = chunk.toString().split(os.EOL);
+      this.chunkToSend += splittedChunk.shift();
+      this.emit('data', this.chunkToSend);
```

Comment:
аналогично первому комментарию - события нужны внешнему коду чтобы узнавать что происходит со стримом, но напрямую события вызывать нельзя. в данном случае для того, чтобы передать данные дальше, нужно вызвать метод `push`, т.е. `this.push(this.chunkToSend)`

Reply:
Спасибо, исправил. Не совсем понятно, для чего нужен аргумент callback, если this.push(chunk) передает данные?

Reply:
Уже нашел ответ на stackoverflow: 

>  A minimal implementation should call the callback function to indicate that the transformation is done, even if that transformation means discarding the chunk.
> 
> To queue a new chunk, call this.push(chunk)—this can be called as many times as required before the callback() if you have multiple pieces to send on.